common/js/src/logging/log-buffer.js

```diff
@@ -85,26 +85,44 @@ goog.inherits(LogBuffer, goog.Disposable);
 goog.exportSymbol('GoogleSmartCard.LogBuffer', LogBuffer);
 
 /**
- * @return {number}
+ * Attaches the given log buffer to the given logger, by letting the buffer
+ * collect all logs sent to the logger. Additionally, all collected logs are
+ * attributed with the specified document location, which simplifies
+ * interpreting the collected logs.
+ * @param {!goog.log.Logger} logger
+ * @param {string} documentLocation
  */
-LogBuffer.prototype.getCapacity = function() {
-  return this.capacity_;
+LogBuffer.attachBufferToLogger = function(logBuffer, logger, documentLocation) {
+  // Note: It's crucial that we're in a static method and calling a global
+  // function (`goog.log.addHandle()`), because when `logBuffer` comes from a
+  // different page (e.g., the background page), we're dealing with two
+  // instances of Closure Library: one in our page and another in the buffer's
+  // page. We want the current page's instance to register the buffer, since
+  // it's where the `logger`s messages are handled.
+  goog.log.addHandler(
+      logger, logBuffer.onLogRecordObserved_.bind(logBuffer, documentLocation));
 };
 
 goog.exportProperty(
-    LogBuffer.prototype, 'getCapacity', LogBuffer.prototype.getCapacity);
+    LogBuffer, 'attachBufferToLogger', LogBuffer.attachBufferToLogger);
+
+/** @override */
+LogBuffer.prototype.disposeInternal = function() {
+  this.formattedLogsPrefix_.length = 0;
+  this.observers_.length = 0;
+  this.formattedLogsSuffix_.length = 0;
```

Comment:
Actually this should've been `clear()`. Not sure why the compiler didn't complain about writing to a non-existing property in `CircularBuffer`.